docs/source/user_guide.md

```diff
@@ -153,8 +165,190 @@ name: new-project
 environments:
   default:
     - environment.yml
+variables: {}
+commands: {}
+```
+
+## Environment variables
+
+Conda Project supports defining environment variables in the `conda-project.yml` file that will
+be set upon environment activation or when running commands. Variables are defined in the
+`variables:` key and can have an optional default value. Starting from the minimal example above
+I'll add a variable `FOO` with the default value `has-default-value` and the variable `BAR` with
+no default value.
+
+```yaml
+name: new-project
+environments:
+  default:
+    - environment.yml
+
+variables:
+  FOO: has-default-value
+  BAR:
+
+commands: {}
+```
+
+The values of the variables can be set or overridden by the use of a `.env` file (see the
+[python-dotenv documentation](https://saurabh-kumar.com/python-dotenv/#file-format) for more details) or
+by setting the variable in your shell.
+
+In this example the `BAR` environment variable is unset so `conda project activate` will not start
```

Comment:
Not sure I would use env vars without default values, but this is a great way to make sure they get defined before use!